staging/src/kubevirt.io/api/clone/v1alpha1/types.go

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VirtualMachineClone is a CRD that clones one VM into another.
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +genclient
+type VirtualMachineClone struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   VirtualMachineCloneSpec   `json:"spec" valid:"required"`
+	Status VirtualMachineCloneStatus `json:"status,omitempty"`
+}
+
+type CloneIncludeExclude struct {
+	// +optional
+	CloneAllByDefault *bool `json:"cloneAllByDefault,omitempty"`
+	// +optional
+	// +listType=atomic
+	Exclude []string `json:"exclude,omitempty"`
+	// +optional
+	// +listType=atomic
+	Include []string `json:"include,omitempty"`
+}
+
+type VirtualMachineCloneSpec struct {
+	Source *corev1.TypedLocalObjectReference `json:"source"`
+
+	// +optional
+	Target *corev1.TypedLocalObjectReference `json:"target,omitempty"`
+
+	// +optional
+	Annotations *CloneIncludeExclude `json:"annotations,omitempty"`
+	// +optional
+	Labels *CloneIncludeExclude `json:"labels,omitempty"`
```

Comment:
Working with these include/exclude options is maybe a little awkward, but fine.  If they are nil, no annotations/labels will be copied, right?  I wonder about supporting the case where a user wants to label/annotate the VM with annotations not on the source.  Do we care about that?

Reply:
> Working with these include/exclude options is maybe a little awkward, but fine

Have any other ideas?...
@davidvossel [suggested another approach](https://github.com/kubevirt/kubevirt/pull/7336#discussion_r888428418), you're welcome to give us your opinion there.

> If they are nil, no annotations/labels will be copied, right?

Currently it's the other way around - all labels would be copied by default. This sounds sensible to me, since IMHO the default should be "clone everything", and changes to the clone would be possible via configs.

> I wonder about supporting the case where a user wants to label/annotate the VM with annotations not on the source. Do we care about that?

I don't think so.
The reason is that more labels / annotations can be added to any VM, so they should be added after the VM is cloned. In my eyes this isn't a part of the cloning operation, but a complementary operation for post-cloning. Does it make sense to you?